translate/storage/cpo.py

```diff
@@ -377,7 +379,11 @@ def setsource(self, source):
         else:
             gpo.po_message_set_msgid(self._gpo_message, gpo_encode(source))
             gpo.po_message_set_msgid_plural(self._gpo_message, None)
-    source = property(getsource, setsource)
+
+    # Deprecated on 2.3.1
+    @deprecated("Use `source` property instead")
+    def getsource(self):
+        return self.source
```

Comment:
Isn't is enough to have this on base.TranslationUnit?

Reply:
Not all the subclasses of `base.TranslationUnit` have the `getsource` and I didn't want to add a new deprecated `getsource` method to those.